chopper.py

```diff
@@ -138,6 +168,44 @@ def parse_args() -> dict[str, Any]:
     return args
 
 
+## [CG] I see why you're reading the whole file into memory in shuffle_file,
+##      but something else you could do is scan for newlines and shuffle their
+##      their byte positions (example below). This is definitely slower, but
+##      probably not by much with your OS buffering your I/O, and you won't
+##      need to allocate a buffer the size of the file in order to shuffle it.
+##
+##      Sadly it's a lot more convoluted than your method. May not be worth it.
+def shuffle_file2(filepath: Path, shuffles: int, config: Config) -> list[Path]:
```

Comment:
This is cool and definitely sped things up. I'm implementing something along these lines in the next version.